onlyswaps-verifier/src/main.rs

```diff
@@ -5,94 +5,130 @@ mod signing;
-                    request_id = verification.request_id.to_string(),
-                    "processing pending verification",
-                );
-                let verified_swap = signer.try_sign(&verification).await.unwrap();
-                tracing::debug!(
-                    chain_id = verification.chain_id,
-                    request_id = verification.request_id.to_string(),
-                    "message signed successfully"
-                );
-                match network_bus
-                    .as_ref()
-                    .submit_verification(verification.chain_id, verified_swap)
-                    .await
-                {
-                    Ok(_) => tracing::info!(chain_id = verification.chain_id, "successful swap"),
-                    Err(e) => tracing::error!(
-                        chain_id = verification.chain_id,
-                        request_id = verification.request_id.to_string(),
-                        error = e.to_string(),
-                        "error submitting verification"
-                    ),
-                }
-            }
+    let EventManagement {
+        event_ids,
+        omnievent,
+    } = create_omnievent_management(&app_config.networks).await?;
+    let mut stream = omnievent.get_ethereum_multi_event_stream(event_ids).await?;
+    let tx1 = tx.clone();
+    tokio::spawn(async move {
+        while let Some(Ok(event)) = stream.next().await {
+            let verification = event.data.try_into().expect("invalid event received");
+            tx1.send(verification)
+                .expect("failed to send verification on channel");
         }
     });
 
+    let tx = tx.clone();
+    for verification in fetch_pending_requests(&app_config).await? {
+        tx.send(verification)?;
+    }
+
+    tracing::info!(
+        port = app_config.agent.healthcheck_port,
+        "Healthcheck server started"
+    );
     let signals = SignalManager::new(
         app_config.agent.healthcheck_listen_addr,
         app_config.agent.healthcheck_port,
     )
     .await?;
-    tracing::debug!(
-        port = app_config.agent.healthcheck_port,
-        "Healthcheck server started"
-    );
-    tokio::select! {
-        signal = signals.next() => {
-            match signal {
-                SignalEvent::SigTerm | SignalEvent::SigInt | SignalEvent::CtrlC => Ok(()),
-                SignalEvent::HealthcheckServerFailed => Err(anyhow!("axum stopped unexpectedly"))
+    let shutdown = signals.next();
+    tokio::pin!(shutdown);
+    loop {
+        tokio::select! {
+            next_verification = rx.recv() => {
+                if let Some(verification) = next_verification {
+                    tracing::info!(
+                        chain_id = verification.chain_id,
+                        request_id = verification.request_id.to_string(),
+                        "attempting verification"
+                    );
+                    let max_timeout = Duration::from_secs(5);
```

Comment:
could consider moving this to the config somewhere (and we might want it configurable by chain [5s on polygon is a lifetime, 5s on filecoin won't cut it])